lib/AST/Type.cpp

```diff
@@ -4098,17 +4112,17 @@ bool TypeBase::usesNativeReferenceCounting(ResilienceExpansion resilience) {
     assert(archetype->requiresClass() ||
            (layout && layout->isRefCounted()));
     if (auto supertype = archetype->getSuperclass())
-      return supertype->usesNativeReferenceCounting(resilience);
-    return ::doesOpaqueClassUseNativeReferenceCounting(type->getASTContext());
+      return supertype->getReferenceCounting(resilience);
+    return getUnknownObjectReferenceCounting();
```

Comment:
In theory we could have a layout constraint but I guess those aren't fully plumbed through yet anyway...